src/components/Board.js

```diff
@@ -4,17 +4,21 @@ import Square from './Square';
 import PropTypes from 'prop-types';
 
 
-const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
+const generateSquareComponents = (squares, onClickCallback, currentPlayer) => {
+  const oneRow = squares.map ((row) => {
+    return row.map ((square) => {
+      return (
+        <Square id={square.id} value={square.value} onClickCallback={onClickCallback} currentPlayer={currentPlayer} />
```

Comment:
Note the following warning. You should add a `key` prop to your Square `key = {square.id`:
   
   ```
     Warning: Each child in a list should have a unique "key" prop. See https://reactjs.org/link/warning-keys for more information.
          at div
          at Board (/app/src/components/Board.js:19:18)
          ```